### PR TITLE
add the ability to specify arbitrary cli options to the analysis server

### PR DESCRIPTION
**Pull request by devoncarew**
- add the ability to specify arbitrary cli options to the analysis server
- fix https://github.com/dart-atom/dartlang/issues/617

@danrubel @bwilkerson

### CODE REVIEW

**Comment by danrubel** [comment]
LGTM

File: lib/analysis_server.dart [hunk 1]
```diff
@@ -546,14 +547,27 @@ class _AnalysisServerWrapper extends Server {
...
+    if (AnalysisServer.startWithDiagnostics) {
       arguments.add('--port=${AnalysisServer.DIAGNOSTICS_PORT}');
       _logger.info('analysis server diagnostics available at '
           '${AnalysisServer.diagnosticsUrl}.');
     }
 
+    // Start in checked mode?
     if (AnalysisServer.useChecked) {
-      arguments.insert(0, '--checked');
+      arguments.add('--checked');
```
Comment (bwilkerson) [inline lib/analysis_server.dart]: I think this change is wrong. The `--checked` is a VM argument, so it really does need to come before the other arguments. (Similar issue in entry.dart.js.)

**Comment by bwilkerson** [comment]
LGTM

File: lib/analysis_server.dart (hunk 1 above)
Comment (devoncarew) [inline lib/analysis_server.dart]: Thanks; updated.

**Comment by devoncarew** [comment]
In order to pass arbitrary options to the analysis server, edit your `config.cson` file (Atom > Open your Config`), and in the `dartlang` section, add a value for `analysisServerOptions`. This can either be a string with space separated values for separate options, or a cson array `["--foo", "--bar"]`.

**Pull request merged**